src/app/api/v1/admin/feature-flags/route.ts

```diff
@@ -28,7 +27,11 @@ export async function GET() {
       return NextResponse.json({ success: false }, { status: 500 });
     }
   } else {
-    return NextResponse.json({ success: false }, { status: 401 });
+    const flagNamesInDb = await getEnabledFeatureFlags({ isSignedOut: true });
+    const availableFlags = featureFlagNames;
+    return NextResponse.json(
+      availableFlags.filter((flag) => flagNamesInDb.indexOf(flag) !== -1),
```

Comment:
I think this function could use a comment to explain the difference if logged in as an admin or not.

Should we consider making the `isAdmin` function a little easier to use, and maybe have some way to ensure that it's used totally consistently? I kind of wish everything denied access by default unless we allow-listed it, instead of the other way around, at least in the `admin` directory.

Reply:
Hmm, so I added a comment in 65b034ed5, but you're right, it's pretty weird to vary the response structure based on whether the user is an admin or not. Of course this is not a public API, and I don't think there are any places that call this API yet, so it's not a pressing issue, but it's not great either way.

We could add middleware that intercepts `/api/v1/admin/` requests and checks whether the user is an admin, and then move this API outside of that namespace. But not super high priority IMO.